src/peft/pipelines/text_generation.py

```diff
@@ -0,0 +1,174 @@
+    peft_model_class = PeftModelForCausalLM
+    task_type = "text-generation"
+    supported_extra_args = {"merge_model": bool, "adapter_name": str}
+
+    def __init__(self, *args, **kwargs):
+        self.merge_model = kwargs.pop("merge_model", False)
+        self.adapter_name = kwargs.pop("adapter_name", "default")
+
+        super().__init__(*args, **kwargs)
+
+    def _load_model_and_processor(self):
+        r"""
+        Load the model and processor for the pipeline. This method first loads the peft config, infers the base model
+        name or path from it and then loads the transformers model and processor.
+        """
+        if isinstance(self.model, str):
+            peft_config = PeftConfig.from_pretrained(self.model)
+            base_model_path = peft_config.base_model_name_or_path
+
+            transformers_model = self.transformers_model_class.from_pretrained(
+                base_model_path, **self.base_model_kwargs
+            )
+
+            if self.processor is None:
+                try:
+                    self.processor = self.transformers_processor_class.from_pretrained(base_model_path)
+                except EntryNotFoundError:
+                    raise ValueError(
+                        f"We couldn't find a `transformers` tokenizer in {base_model_path} - make sure this is a correct path or url to a model checkpoint."
+                    )
+            elif not isinstance(self.processor, self.transformers_processor_class):
+                raise ValueError(
+                    f"The provided `tokenizer` does not match the associated model - got {self.processor.__class__} whereas it should be {self.transformers_processor_class}"
+                )
+
+            self.model = self.peft_model_class.from_pretrained(
+                transformers_model,
+                self.model,
+                adapter_name=self.adapter_name,
+            )
+
+            if self.device is None and not hasattr(self.model, "hf_device_map"):
+                self.device = self.model.base_model.device
+
+            if getattr(self, "merge_model", False):
+                self.model = self.model.merge_and_unload()
+                self.merged_model = True
+            else:
+                self.merged_model = False
+
+        elif isinstance(self.model, self.peft_model_class):
+            self.processor = self.transformers_processor_class.from_pretrained(
+                self.model.peft_config.base_model_name_or_path
+            )
+        else:
+            raise ValueError(
+                f"The model must be a path to a checkpoint or a {self.peft_model_class.__class__} instance, got {self.model.__class__}"
+            )
+
+        if self.processor.pad_token is None:
+            self.processor.pad_token = self.processor.eos_token
+
+    def __call__(self, text, **kwargs):
+        r"""
+        Generate the text give text inputs using the underlying PEFT model. For LoRA models only it is possible to
```

Comment:
The sentence is not quite correct, did you mean "give" => "given"?

Reply:
Yes correct!